Book-eCommerce-Store/Services/OrdersService/OrdersService.cs

```diff
@@ -69,13 +74,55 @@ public async Task<Response> CreateOrder(CreateOrderDTO newOrder)
                     mappedToPurchase.ProductId = mappedToProduct.Id;
                     order.subtotalInCent+=mappedToPurchase.PriceInCent*mappedToPurchase.Quantity;
                     order.purchasedProducts.Add(mappedToPurchase);
+
+                    DiscountName = mappedToPurchase.Name;
+                    DiscountQuantityCount = mappedToPurchase.Quantity;
+                    DiscountCategory = (int)mappedToPurchase.ProductCategory;
                 }
```

Comment:
Think you accidently placed the code to calculate the discount outside the foreach loop in CreateOrder. At the moment it will only apply a discount to the last item in the list if applicable because DiscountName, DiscountQuantity and DiscountCategory will keep getting overwritten.

Reply:
I should have it sorted now, if something else needs to be changed lemme know